### ISSUE TITLE
Segfault While Tracing Method

### ISSUE THREAD

**Issue by obsh** [census-instrumentation/opencensus-php#183]
Hi, I’m experiencing segfault with extension in my Symfony project on Ubuntu Xenial and Debain Jessie.
phpinfo() reports that extension version is 0.2.1. Compiled from release v0.5.0.

During debugging I’ve discovered that error disappear if iI avoid tracing of one particular method,
tracing of other methods works fine.

```php
        // public function publish(string $topicName, string $data, array $attributes = [])
        opencensus_trace_method(QueueService::class, 'publish', function ($queue, $topicName) {
            return [
                'name' => 'queue.publish',
                'attributes' => [
                    'topicName' => $topicName
                ]
            ];
        });
```

Also, there is no segfault if I skip callback:

```php
        // public function publish(string $topicName, string $data, array $attributes = [])
        opencensus_trace_method(QueueService::class, 'publish');
```

It’s 100% reproducible on my project, but I’m not sure which minimal test case can reveal this issue.

With gdb attached to PHP-FPM process I was able to get this backtrace:

```
Thread 1 "php-fpm7.1" received signal SIGSEGV, Segmentation fault.
0x000056251320dc12 in _emalloc_56 ()
(gdb) bt
#0  0x000056251320dc12 in _emalloc_56 ()
#1  0x0000562513237e70 in _array_init ()
#2  0x0000562513250ce8 in zend_fetch_debug_backtrace ()
#3  0x00007f105478e020 in opencensus_trace_begin (name=name@entry=0x7f1054ce02a0, span_id=span_id@entry=0x0,
    execute_data=0x7f1054c133f0) at /opencensus-php-0.5.0/ext/opencensus_trace.c:400
#4  0x00007f105478e4d0 in opencensus_trace_execute_ex (execute_data=0x7f1054c133f0)
    at /opencensus-php-0.5.0/ext/opencensus_trace.c:622
#5  0x00005625132d898a in ?? ()
#6  0x000056251327f92b in execute_ex ()
#7  0x00007f105478e561 in opencensus_trace_execute_ex (execute_data=0x7f1054c13340)
    at /opencensus-php-0.5.0/ext/opencensus_trace.c:627
#8  0x00005625132d851e in ?? ()
#9  0x000056251327f92b in execute_ex ()
#10 0x00007f105478e561 in opencensus_trace_execute_ex (execute_data=0x7f1054c13260)
    at /opencensus-php-0.5.0/ext/opencensus_trace.c:627
#11 0x00005625132d851e in ?? ()
#12 0x000056251327f92b in execute_ex ()
#13 0x00007f105478e4dc in opencensus_trace_execute_ex (execute_data=0x7f1054c131b0)
    at /opencensus-php-0.5.0/ext/opencensus_trace.c:623
#14 0x00005625132d851e in ?? ()
#15 0x000056251327f92b in execute_ex ()
#16 0x00007f105478e561 in opencensus_trace_execute_ex (execute_data=0x7f1054c13110)
    at /opencensus-php-0.5.0/ext/opencensus_trace.c:627
#17 0x00005625132d851e in ?? ()
#18 0x000056251327f92b in execute_ex ()
#19 0x00005625132dab10 in zend_execute ()
#20 0x0000562513235de3 in zend_execute_scripts ()
#21 0x00005625131d20e0 in php_execute_script ()
#22 0x00005625130b0079 in main ()
```

I’m not experienced with extensions development or gdb, so please let me know if I can collect some additional info for resolving this issue.

**Comment by chingor13** [comment]
Thanks for the report! I'll start looking into this today.

**Comment by obsh** [comment]
Sure:
```
root@e14cac96875d:/application# php -v
PHP 7.1.17-1+ubuntu16.04.1+deb.sury.org+1 (cli) (built: May  5 2018 04:55:21) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.1.17-1+ubuntu16.04.1+deb.sury.org+1, Copyright (c) 1999-2018, by Zend Technologies
```

**Comment by chingor13** [comment]
@obsh I was unable to reproduce this exact case, but I found a bug that I think is the cause - we were not copying the argument values before executing the original function - meaning the argument values could be deallocated by the memory manager before the callback runs. Working on this in #184

**Comment by obsh** [comment]
I've built extension from branch 7.1-segfault and tested my use case - no segfault. Thank you!

**Comment by chingor13** [comment]
This should be available on PECL as 0.2.2 shortly.
